### ISSUE TITLE
NNBD support for part generation

### ISSUE THREAD

**Issue by davidmorgan** [dart-lang/source_gen#473]
We're going to need `source_gen` to work while generating into libraries that are migrating to NNBD :)

Generators will need to be able to detect NNBD status of the library they are generating for. As far as I can tell, this is up to the analyzer and `package:build`; I'll file an issue at `package:build`.

And, the generators in `source_gen` that generate part files will need to match the language version comment in the target library.

I was able to hack something that "works" by checking for a language comment in the input library source and matching it; in `lib/builder.dart`:

```dart
    final library = await buildStep.inputLibrary;
    final source = library.source.contents.data;
    final version = source.contains('// @dart=2.9')
        ? '\n// @dart=2.9'
        : source.contains('// @dart=2.8') ? '\n// @dart=2.8' : '';

    final ignoreForFile = _ignoreForFile.isEmpty
        ? ''
        : '\n// ignore_for_file: ${_ignoreForFile.join(', ')}\n';
    final output = '''
$defaultFileHeader
$ignoreForFile$version
part of $partOf;
```

Instead of checking the source we may be able to use `library.isNonNullableByDefault`, but currently that seems to always return false--which is the `package:build` issue, it needs to understand NNBD config.

**Comment by scheglov** [comment]
See `LibraryElement.languageVersion` - it allows you to know if the library has a language version override, so its parts should as well.

**Comment by kevmoo** [comment]
Fixed in cd8a9d0eab454ee61acdd93e294e2c0c8463686e – published as https://github.com/dart-lang/source_gen/releases/tag/v0.9.6
